vendors/circom/prover_main.cc

```diff
@@ -75,13 +75,22 @@ void CreateProof(const base::FilePath& zkey_path,
       QuadraticArithmeticProgram<F>::WitnessMapFromMatrices(
           domain.get(), constraint_matrices, full_assignments);
 
-  zk::r1cs::groth16::Proof<Curve> proof =
-      zk::r1cs::groth16::CreateProofWithAssignmentNoZK(
-          proving_key, absl::MakeConstSpan(h_evals),
-          full_assignments.subspan(
-              1, constraint_matrices.num_instance_variables - 1),
-          full_assignments.subspan(constraint_matrices.num_instance_variables),
-          full_assignments.subspan(1));
+  zk::r1cs::groth16::Proof<Curve> proof;
+  if (!no_zk) {
+    proof = zk::r1cs::groth16::CreateProofWithAssignmentZK(
+        proving_key, absl::MakeConstSpan(h_evals),
+        full_assignments.subspan(
+            1, constraint_matrices.num_instance_variables - 1),
+        full_assignments.subspan(constraint_matrices.num_instance_variables),
+        full_assignments.subspan(1));
+  } else {
+    proof = zk::r1cs::groth16::CreateProofWithAssignmentNoZK(
+        proving_key, absl::MakeConstSpan(h_evals),
+        full_assignments.subspan(
+            1, constraint_matrices.num_instance_variables - 1),
+        full_assignments.subspan(constraint_matrices.num_instance_variables),
+        full_assignments.subspan(1));
+  }
```

Comment:
negation of boolean expression is hard to read. Please change as follows!

```suggestion
  if (no_zk) {
    proof = zk::r1cs::groth16::CreateProofWithAssignmentNoZK(
        proving_key, absl::MakeConstSpan(h_evals),
        full_assignments.subspan(
            1, constraint_matrices.num_instance_variables - 1),
        full_assignments.subspan(constraint_matrices.num_instance_variables),
        full_assignments.subspan(1));
  } else {
    proof = zk::r1cs::groth16::CreateProofWithAssignmentZK(
        proving_key, absl::MakeConstSpan(h_evals),
        full_assignments.subspan(
            1, constraint_matrices.num_instance_variables - 1),
        full_assignments.subspan(constraint_matrices.num_instance_variables),
        full_assignments.subspan(1));
  }
```